### PR TITLE
(CODEMGMT-440) Update shellgit thin repo alternates

### PR DESCRIPTION
**Pull request by adrienthebo**
This defers syncing the alternates file till the first time the Git command is run so the file is only updated when needed.

### CODE REVIEW

File: lib/r10k/git/shellgit/thin_repository.rb [hunk 1]
```diff
@@ -51,4 +45,21 @@ def setup_cache_remote
     git ["remote", "add", "cache", @cache_repo.git_dir.to_s], :path => @path.to_s
     fetch
   end
+
+  def git(cmd, opts = {})
+    if @_synced_alternates
```
Comment (scotje) [inline lib/r10k/git/shellgit/thin_repository.rb]: Should this be if-not or unless?

Comment (adrienthebo) [inline lib/r10k/git/shellgit/thin_repository.rb]: Yep, good catch.

**Comment by scotje** [comment]
:+1: Looks good to me

**Comment by adrienthebo** [comment]
We're going to ship this in r10k 2.1.2, so holding off on merging it till we ship 2.1.1.

**Comment by scotje** [comment]
We have a dependent PR which needs to go in 2.1.1 so we might as well include this.

**Pull request merged**